### PR TITLE
Update dependencies

### LINKED ISSUE #50: Update to url 1.x
With rust 1.11.0 (stable) cargo warns about deprecated Cargo.toml syntax in the url 0.5.9 crate.

```
warning: TOML file found which contains invalid syntax and will soon not parse
at `/home/giles/.cargo/registry/src/github.com-1ecc6299db9ec823/url-0.5.9/Cargo.toml`.

The TOML spec requires newlines after table definitions (e.g. `[a] b = 1` is
invalid), but this file has a table header which does not have a newline after
it. A newline needs to be added and this warning will soon become a hard error
in the future.
```

Racerd doesn't depend on the (now stable) url crate directly but it does through hyper and iron. Both have had minor revision bumps to update to the 1.x release of the crate, but other changes mean some code changes are necessary to complete the update.

### PR DESCRIPTION
**Pull request by jwilm**
Resolves #50, #48, #45.

---
This change is [Reviewable](https://reviewable.io/reviews/jwilm/racerd/53)

### CODE REVIEW

File: src/http/definition.rs.in [hunk 1]
```diff
@@ -105,7 +105,7 @@ fn find_definition_request_from_json() {
     assert_eq!(req.column, 3);
 }
 
-#[derive(Debug, Deserialize, Serialize)]
+#[derive(Debug, Serialize, Deserialize)]
```
Comment (rillian) [inline src/http/definition.rs.in]: Is the order important? I don't think I would ever have thought of that!

Comment (jwilm) [inline src/http/definition.rs.in]: It doesn't actually matter. This was an attempt to fix a weird error that turned out to be due to an old version of serde_codegen. I'll revert this line.

Comment (rillian) [inline src/http/definition.rs.in]: Ok, thanks for explaining!

**Comment by rillian** [comment]
I filed https://github.com/contain-rs/vec-map/pull/22 about the vec_map issue.

**Comment by jwilm** [comment]
Thanks! I wonder if that was a bug in beta? Current nightly doesn't seem to have that issue.

**Pull request merged**